pages/apim/3.x/v4/v4-introduction.adoc

```diff
@@ -0,0 +1,24 @@
+[[v4-introduction]]
+= V4 policy execution engine (V4 BETA)
+:page-sidebar: apim_3_x_sidebar
+:page-permalink: apim/3.x/v4_introduction.html
+:page-folder: apim/event-native-apim
+:page-layout: apim3x
+
+[label label-version]#New in version 3.20.0#
+[label label-version]#BETA release#
+
+Introduced in release 3.19.0 of the Gravitee API Management platform, there is a beta version of a new policy execution engine that enables an improved execution flow for synchronous APIs and supports event-driven policy execution for asynchronous APIs. It is based on a modern and fully reactive architecture designed to address a number of challenges Gravitee users have been facing with the existing policy execution engine, available with the Gravitee's V2 API definition specification used in versions prior to APIM 3.19.x.
+
+The new policy execution engine builds on the new V4 BETA API definition foundation of Gravitee's event-native platform, which adds features such as native support for Pub/Sub (Publish-Subscribe) design and enabling policies at message level. The new V4 BETA event-native API definition makes it possible for the Gateway to “know” what is necessary in order for it to handle asynchronous API use cases.
+
+The new engine provides the following capabilities:
+
+* The ability to execute policies in the exact order in which they have been placed in Policy Studio. This addresses some issues experienced by users related to the order in which policies are executed by the V3 engine where policies interacting with the Head part of the request are always executed first, even when placed in a different order in the Policy Studio during the design phase. With the new execution engine, it is possible to apply logic on a head policy based on the payload of the request - for example, to apply dynamic routing based on the request payload.
```

Comment:
```suggestion
* The ability to execute policies in the exact order in which they have been placed in Design Studio. This addresses some issues experienced by users related to the order in which policies are executed by the V3 engine where policies interacting with the Head part of the request are always executed first, even when placed in a different order in the Design Studio during the design phase. With the new execution engine, it is possible to apply logic on a head policy based on the payload of the request - for example, to apply dynamic routing based on the request payload.
```

Reply:
@mouligno we mean Design Studio here, right?
@AndrewMcFarlandCampbellGravitee we could add the xref link to the relevant doc pages e.g. https://docs.gravitee.io/apim/3.x/apim_publisherguide_design_studio_create.html#flow-policies and https://docs.gravitee.io/apim/3.x/apim_publisherguide_plan_policies.html